src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -1713,6 +1943,216 @@ LogicalResult ONNXSplitOp::inferShapes() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// Flatten
+
+LogicalResult ONNXFlattenOp::inferShapes() {
+  assert(axis() == 1 && "ONNXFlattenOp can only handle axis=1 for now");
+  auto inTy = input().getType().dyn_cast<ShapedType>();
+  if (!inTy) {
+    return emitOpError("Input is a non-shaped type");
+  }
+  auto outTy = output().getType().dyn_cast<ShapedType>();
+  if (!outTy) {
+    return emitOpError("Output is a non-shaped type");
+  }
+
+  if (!outTy.hasStaticShape()) {
```

Comment:
Seems like we can also fairly easily support the case where the batch dimension is dynamic. Feel free to leave a todo note for me if this does not concern you.